packages/synapse-react-client/test/lib/containers/AccessRequirementList.test.tsx

```diff
@@ -0,0 +1,73 @@
+import { act, render, screen, waitFor } from '@testing-library/react'
+import * as React from 'react'
+import AccessRequirementListV2, {
+  AccessRequirementListProps,
+} from '../../../src/lib/containers/AccessRequirementListV2/AccessRequirementListV2'
+import { createWrapper } from '../../testutils/TestingLibraryUtils'
+import { AccessRequirement } from '../../../src/lib/utils/synapseTypes/'
+import mockFileEntityData from '../../../mocks/entity/mockFileEntity'
+import { server } from '../../../mocks/msw/server'
+import {
+  mockACTAccessRequirement,
+  mockManagedACTAccessRequirement,
+  mockSelfSignAccessRequirement,
+  mockToUAccessRequirement,
+} from '../../../mocks/mockAccessRequirements'
+import { SynapseClient } from '../../../src/lib'
+
+const MOCK_FILE_ENTITY_ID = mockFileEntityData.id
+
+const sortAccessRequirementsByCompletionSpy = jest.spyOn(
+  SynapseClient,
+  'sortAccessRequirementsByCompletion',
+)
+
+describe('AccessRequirementList tests', () => {
+  beforeAll(() => server.listen())
+  afterEach(() => server.restoreHandlers())
+  afterAll(() => server.close())
+
+  async function init(props: AccessRequirementListProps) {
+    await act(() => {
+      render(<AccessRequirementListV2 {...props} />, {
+        wrapper: createWrapper(),
+      })
+    })
+  }
+
+  const accessRequirements: Array<AccessRequirement> = [
+    mockSelfSignAccessRequirement,
+    mockToUAccessRequirement,
+    mockManagedACTAccessRequirement,
+    mockACTAccessRequirement,
+  ]
+
+  const props: AccessRequirementListProps = {
+    entityId: MOCK_FILE_ENTITY_ID,
+    accessRequirementFromProps: accessRequirements,
+    onHide: jest.fn(),
+  }
+
+  it('Renders a Access Requirements List with valid props', async () => {
```

Comment:
This isn't much more than the original set of tests. It would help to add more integration tests but they might be flaky.

Reply:
I want to add more but I also want to finish this feature. I added many unit tests to compensate.